src/service/product.service.ts

```diff
@@ -96,4 +98,59 @@ export class ProductsService {
     });
     return productExist;
   }
+  public async updateProduct(params: ProductAdminDto) {
+    try {
+      const id = params?.productId;
+      const product = await this.productRepository.findOne({
+        where: { id: undefined },
+      });
+
+      if (product && id !== undefined) {
+        const productUpdated = this.productRepository.create({
+          id: product.id,
+          ...params,
+        });
+        await this.productRepository.save(productUpdated);
+      } else {
+        const productUpdated = this.productRepository.create({
+          ...params,
+        });
+        await this.productRepository.save(productUpdated);
+      }
+
+      return product;
+    } catch (error) {
+      console.log('Error update product', error);
+      return null;
+    }
+  }
+
+  public async deleteProduct(id: number) {
+    try {
+      const product = await this.productRepository.findOne({
+        where: { id },
+      });
+      if (product) {
+        await this.productRepository.delete({ id });
```

Comment:
khi xóa product thì phải check các logic liên quan e nhé
vd: nếu sp đã đc user đặt rồi thì có cho xóa hay ko?

Reply:
Dạ em nghĩ solution trong trường hợp này là em xóa mềm product ( thêm field active ) để toàn vẹn những dữ liệu như hóa đơn ạ